### PR TITLE
chore: extract complex if condition to improve readability 

### PR DESCRIPTION
**Pull request by anikethsaha**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
NA

### Motivation / Use-Case
https://dzone.com/articles/code-smells-if-statements

To improve the readablilty of the code, extracted most of the complex or huge if conditions. 
Tried to cover as much as possible, as of now

### Breaking Changes
NO

### Additional Info

NA

### CODE REVIEW

**Comment by anikethsaha** [comment]
> Let's use better name for variables, for example `shouldParsePath`,`isHostname`, `shouldEmit` (maybe better name) 👍

Sure !... yeah these are better name 👍

**Comment by anikethsaha** [comment]
cc @evilebottnawi

File: lib/Server.js [hunk 1]
```diff
@@ -889,23 +894,18 @@ class Server {
...
           }
         }
       }
     }
 
-    // allow hostname of listening address
-    if (hostname === this.hostname) {
-      return true;
-    }
-
```
Comment (evilebottnawi) [inline lib/Server.js]: Why we remove this?

Comment (anikethsaha) [reply lib/Server.js]: I added this above `hostname === this.hostname;` Line :879

File: lib/Server.js [hunk 2]
```diff
@@ -870,13 +870,18 @@ class Server {
...
     // these are removed from the hostname in url.parse(),
     // so we have the pure IPv6-address in hostname.
-    if (ip.isV4Format(hostname) || ip.isV6Format(hostname)) {
+    // always allow localhost host, for convenience (hostname === 'localhost')
+    // allow hostname of listening address  (hostname === this.hostname)
+    const isHostname =
+      ip.isV4Format(hostname) ||
+      ip.isV6Format(hostname) ||
+      hostname === 'localhost' ||
+      hostname === this.hostname;
```
Comment (evilebottnawi) [inline lib/Server.js]: Let's renamr to `isValidHostname` :+1:

File: lib/Server.js (hunk 1 above)
Comment (evilebottnawi) [reply lib/Server.js]: :+1:

File: lib/Server.js (hunk 2 above)
Comment (anikethsaha) [reply lib/Server.js]: sure ... 👍

Comment (anikethsaha) [reply lib/Server.js]: done 👍

**Pull request merged**